kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,462 @@
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.freedesktop.dbus.errors.NotSupported;
+import org.junit.Test;
+
+public class NetworkPropertiesTest {
+
+	NetworkProperties netProps;
+	Map<String, Object> properties = new HashMap<String, Object>();
+	Optional<?> optResult;
+	String stringResult; // Boolean, Short, Integer, Long
+	Boolean booleanResult;
+	Short shortResult;
+	int intResult;
+	Long longResult;
+
+	List<String> stringListResult;
+	Map<String, Object> resultMap;
+
+	Boolean hasNullPointExceptionBeenThrown = false;
+	Boolean hasNoSuchElementExceptionBeenThrown = false;
+
+	@Test
+	public void shouldFailWhenNull() {
+		givenNetworkPropertiesBuiltWith(null);
+		thenANullPointerExceptionOccured();
+	}
+
+	@Test
+	public void shouldFailWhenEmpty() {
+		givenNetworkPropertiesBuiltWith(new HashMap<String, Object>());
+		whenGetPropertiesIsCalled();
+		thenNoExceptionsOccured();
+		thenMapResultEquals(new HashMap<String, Object>());
+	}
+
+	@Test
+	public void shouldReturnPropertiesWithGetProperties() {
+		givenMapWith("testKey1", "testString1");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetPropertiesIsCalled();
+		thenNoExceptionsOccured();
+		thenMapResultEquals(Collections.singletonMap("testKey1", "testString1"));
+	}
+
+	@Test
+	public void shouldReturnStringWithGet() {
+		givenMapWith("testKey1", "testString1");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetIsCalledWith("testKey1", String.class);
+		thenNoExceptionsOccured();
+		thenStringResultEquals("testString1");
+	}
+
+	@Test
+	public void shouldReturnStringWithGetNull() {
+		givenMapWith("testKey1", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetIsCalledWith("testKey1", String.class);
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnStringWithGetMissing() {
+		givenMapWith("testKey1", null);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetIsCalledWith("testKey1-nonExistant", String.class);
+		thenANoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnStringWithGetEmptyString() {
+		givenMapWith("", "");
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetIsCalledWith("", String.class);
+		thenNoExceptionsOccured();
+		thenStringResultEquals("");
+	}
+	
+	@Test
+	public void shouldReturnBooleanWithGet() {
+		givenMapWith("testKey1", false);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetIsCalledWith("testKey1", Boolean.class);
+		thenNoExceptionsOccured();
+		thenBooleanResultEquals(false);
+	}
+	
+	@Test
+	public void shouldReturnShortWithGet() {
+		givenMapWith("testKey1", (short) 10);
+		givenNetworkPropertiesBuiltWith(this.properties);
+		whenGetIsCalledWith("testKey1", Short.class);
+		thenNoExceptionsOccured();
+		thenShortResultEquals((short) 10);
```

Comment:
Use class constructors instead of casting

```suggestion
		thenShortResultEquals(new Short(10));
```

Reply:
are you sure? some of the constructors are deprecated? 

IE
<img width="218" alt="image" src="https://user-images.githubusercontent.com/29900100/217011103-45bee51e-2bc2-469b-aa56-ff85d5aa426a.png">

Reply:
<img width="359" alt="image" src="https://user-images.githubusercontent.com/29900100/217014754-7bc8d486-2673-4ff8-86ba-3000308570ed.png">

Reply:
<img width="359" alt="image" src="https://user-images.githubusercontent.com/29900100/217014754-7bc8d486-2673-4ff8-86ba-3000308570ed.png">

Reply:
They are deprecated since version 9. Will this be an issue for java 17?